driver/reducecalculation_driver.hpp

```diff
@@ -136,18 +173,20 @@ class ReduceCalculationDriver : public Driver
     std::unique_ptr<GPUMem> workspace_dev;
 
     std::vector<Tgpu> in;
-    std::vector<Tgpu> out;
+    std::vector<Tgpu_out> out;
```

Comment:
```suggestion
    std::vector<Tref> out;
```

Reply:
i think it should be `std::vector<Tgpu> out;`, as we copied this value from GPU and `Tref` type for `outhost` (which is calculated on cpu). Please check my new solution